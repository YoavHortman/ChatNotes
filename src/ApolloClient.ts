import {ApolloClient} from 'apollo-client';
import {HttpLink} from 'apollo-link-http';
import {InMemoryCache} from 'apollo-cache-inmemory';

export const client = new ApolloClient({
    link: new HttpLink({uri: "https://api.graph.cool/simple/v1/cji5xs8h03jev0167j4k31qf5"}),
    cache: new InMemoryCache(),
});