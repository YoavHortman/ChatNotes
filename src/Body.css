.Body__root {
    display: flex;
    flex-direction: column;
    overflow: scroll;
}