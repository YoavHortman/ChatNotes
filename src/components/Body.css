.Body__root {
    display: flex;
    flex-direction: column;
    overflow: scroll;
}

.Body__messageRoot {
    display: flex;
    background: red;
    flex-direction: column;
    justify-content: stretch;
    margin: 10px;
    flex: 0 0 auto;
}

.Body__messageBody {
    display: flex;
    flex-direction: row;
}

.Body__messageInitialsContainer {
    border-radius: 50%;
    background: blue;
    width: 40px;
    height: 40px;
    display:flex;
    align-items: center;
    justify-content: center;
    font-weight: bold;
}

.Body__messageContainer {
    display: flex;
    flex-direction: row;
    align-items: center;
    background: green;
}

.Body__editIcon {
    background: url("../assets/EditIcon.png");
    background-size: contain;
    width: 30px;
    height: 30px;
}

.Body__XIcon {
    background: url("../assets/XIcon.png");
    background-size: contain;
    width: 30px;
    height: 30px;
}