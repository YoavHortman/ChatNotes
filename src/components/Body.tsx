import * as React from "react";
import "./Body.css"
import {ChatMessage} from "../App";

interface BodyProps {
    messages: ChatMessage[];
    onDeleteClick: (messageId: string) => void;
    onEditClick: () => void;
}

interface BodyState {
}

export class Body extends React.Component<BodyProps, BodyState> {
    constructor(props: BodyProps) {
        super(props);
    }

    handleEditClick = () => {

    }

    getFormattedDateText(date: Date) {
        return `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
    }

    getFormattedTimeText(date: Date) {
        return `${date.getHours()}:${date.getMinutes()}`;
    }

    render() {
        return (
            <div className={"Body__root"}>
                {this.props.messages.sort((message1, message2) => {
                    return message1.createdAt.getTime() - message2.createdAt.getTime();
                }).map(message => {
                    return (<div className={"Body__messageRoot"} key={message.id}>
                        <div className={"Body__messageBody"}>
                            <div className={"Body__messageInitialsContainer"}>
                                {message.createdBy}
                            </div>
                            <div className={"Body__messageContainer"}>
                                <div>{message.text}</div>
                                <div className={"Body__editIcon"} onClick={this.handleEditClick}/>
                                <div className={"Body__XIcon"} onClick={() => this.props.onDeleteClick(message.id)}/>
                            </div>
                        </div>
                        <div className={"Body__messageDataContainer"}>
                            {this.getFormattedDateText(message.createdAt)} {this.getFormattedTimeText(message.createdAt)}
                        </div>
                    </div>);
                })}
            </div>
        );
    }
}
