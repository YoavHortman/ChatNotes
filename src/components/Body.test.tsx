import * as React from "react";
import * as renderer from "react-test-renderer";
import {Body} from "./Body";
import {client} from "../ApolloClient";
import {ApolloProvider} from "react-apollo";

it('Should render an empty page', () => {
    const tree = renderer
        .create(<ApolloProvider client={client}>
            <Body
                messages={[]}
            />
        </ApolloProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Should render the entire list', () => {
    const message1 = {
        text: "Hello!",
        id: "1",
        createdAt: new Date(2012, 11, 25, 5, 53),
        createdBy: "YH"
    };
    const message3 = {
        text: "Test!",
        id: "2",
        createdAt: new Date(2012, 11, 25, 5, 53),
        createdBy: "YH"
    };
    const message2 = {
        text: "Test1!",
        id: "3",
        createdAt: new Date(2012, 11, 25, 5, 53),
        createdBy: "YH"
    };
    const tree = renderer
        .create(<ApolloProvider client={client}>
            <Body
                messages={[message1, message3, message2]}
            />
        </ApolloProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});