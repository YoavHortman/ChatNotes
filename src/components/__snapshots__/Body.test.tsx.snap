// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an empty page 1`] = `
<div
  className="Body__root"
>
  <div />
</div>
`;

exports[`Should render the entire list 1`] = `
<div
  className="Body__root"
>
  <div>
    <div
      className="Message__root"
    >
      <div
        className="Message__body"
      >
        <div
          className="Message__initials"
        >
          YH
        </div>
        <div
          className="Message__messageContainer"
        >
          <div
            className="Message__triangle"
          />
          <div
            className="Message__content"
          >
            <div
              className="Message__text"
            >
              Hello!
            </div>
            <div
              className="Message__iconContainer"
            >
              <div
                className="Message__editIcon"
                onClick={[Function]}
              />
              <div
                className="Message__XIcon"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="Message__footer"
      >
        25/12/2012
         
        5:53
      </div>
    </div>
    <div
      className="Message__root"
    >
      <div
        className="Message__body"
      >
        <div
          className="Message__initials"
        >
          YH
        </div>
        <div
          className="Message__messageContainer"
        >
          <div
            className="Message__triangle"
          />
          <div
            className="Message__content"
          >
            <div
              className="Message__text"
            >
              Test!
            </div>
            <div
              className="Message__iconContainer"
            >
              <div
                className="Message__editIcon"
                onClick={[Function]}
              />
              <div
                className="Message__XIcon"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="Message__footer"
      >
        25/12/2012
         
        5:53
      </div>
    </div>
    <div
      className="Message__root"
    >
      <div
        className="Message__body"
      >
        <div
          className="Message__initials"
        >
          YH
        </div>
        <div
          className="Message__messageContainer"
        >
          <div
            className="Message__triangle"
          />
          <div
            className="Message__content"
          >
            <div
              className="Message__text"
            >
              Test1!
            </div>
            <div
              className="Message__iconContainer"
            >
              <div
                className="Message__editIcon"
                onClick={[Function]}
              />
              <div
                className="Message__XIcon"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="Message__footer"
      >
        25/12/2012
         
        5:53
      </div>
    </div>
  </div>
</div>
`;
